uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/TokenRequestLatch.java

```diff
@@ -0,0 +1,58 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.client.ui.automation;
+
+import androidx.annotation.NonNull;
+
+import org.junit.Assert;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A custom {@link CountDownLatch} to be used during UI Automation Token Requests.
+ */
+public class TokenRequestLatch extends CountDownLatch {
```

Comment:
Can we contain rather than extend here?

Reply:
As in add `CountDownLatch` as a member field to this class?

Reply:
And, for real, you should really look at moving this whole thing to a Future, where you don't need to worry about constructing whichever version of a latch that you want, or if we later want to change our synchronization on this we don't have to change all the code that uses it.

Reply:
Okay, let me take a look into it

Reply:
But extending the java base classes isn't always safe.  We got in trouble doing this this involving HashMap once, and I'm leery of it forevermore.

Reply:
I'll be looking into the synchronization again as part of this work item: https://dev.azure.com/IdentityDivision/Engineering/_boards/board/t/Auth%20Client%20-%20Android/Backlog%20items/?workitem=1184801

(So I only have to do it once for both ADAL / MSAL).

Reply:
See comment above, going to merge this in for now to do some work on running tests on pipeline. Will come back to it.